src/components/PostcodeSearch/index.js

```diff
@@ -25,7 +27,10 @@ function PostcodeSearch() {
 
   useEffect(() => {
     if (!postcode || !postcode.match(POSTCODE_REGEX)) {
+      setSelectedPostcode(null);
```

Comment:
the null postcode issue is probably this

Reply:
if you get rid of this line does anything break?

Reply:
Nothing breaks as far as I can tell. Once validated the postcode remains in the results info until completely removed, as opposed to reverting to null as soon as the regex failed.

Working as intended I would say.

Reply:
yep looks good